gammapy/image/models/core.py

```diff
@@ -171,6 +174,134 @@ def evaluate(lon, lat, lon_0, lat_0, sigma):
+
+    **Caveat:** The model is normalized to 1 on the plane, i.e. in small angle approximation:
+    :math:`N = 1/(2 \pi \sigma_M \sigma_m)`. This means that for huge elongated Gaussians on the sky
+    this model is not correctly normalized. However, this approximation is perfectly acceptable for the more
+    common case of models with modest dimensions: indeed, the error introduced by normalizing on the plane
+    rather than on the sphere is below 0.1\% for Gaussians with radii smaller than ~ 5 deg.
+
+    Parameters
+    ----------
+    lon_0 : `~astropy.coordinates.Longitude`
+        :math:`\text{lon}_0`: `lon` coordinate for the center of the Gaussian.
+    lat_0 : `~astropy.coordinates.Latitude`
+        :math:`\text{lat}_0`: `lat` coordinate for the center of the Gaussian.
+    sigma_semi_major : `~astropy.coordinates.Angle`
+        Length of the major semiaxis of the Gaussian, in angular units.
+    e : `float`
+        Eccentricity of the Gaussian (:math:`0< e< 1`).
+    theta : `~astropy.coordinates.Angle`
+        :math:`\theta`:
+        Rotation angle of the major semiaxis.  The rotation angle increases counter-clockwise
+        from the North direction.
+    frame : {"galactic", "icrs"}
+        Coordinate frame of `lon_0` and `lat_0`.
+
+
+    Examples
+    --------
+    .. plot::
+        :include-source:
+
+        import numpy as np
+        import matplotlib.pyplot as plt
+        import astropy.units as u
+        from astropy.coordinates import Angle
+        from gammapy.image.models.core import SkyEllipse
+        from gammapy.maps import Map, WcsGeom
+
+        m_geom = WcsGeom.create(binsz=0.01, width=(5, 5), skydir=(2, 2), coordsys="GAL", proj="AIT")
+        theta = Angle("30 deg")
+        model = SkyGaussianElongated("2 deg", "2 deg", "1 deg", 0.7, theta, frame='galactic')
+
+
+        coords = m_geom.get_coord()
+        lon = coords.lon * u.deg
+        lat = coords.lat * u.deg
+        vals = model(lon, lat)
+        skymap = Map.from_geom(m_geom, data=vals.value)
+
+        _, ax, _ = skymap.smooth("0.05 deg").plot()
+
+        transform = ax.get_transform('galactic')
+        ax.scatter(2, 2, transform=transform, s=20, edgecolor='red', facecolor='red')
+        ax.text(1.5, 1.85, r"$(l_0, b_0)$", transform=transform, ha="center")
+        ax.plot([2, 2 + np.sin(theta)], [2, 2 + np.cos(theta)], color="r", transform=transform)
+        ax.vlines(x=2, color='r', linestyle='--', transform=transform, ymin=-5, ymax=5)
+        ax.text(2.25, 2.45, r"$\theta$", transform=transform);
+        ax.contour(skymap.data, cmap='coolwarm', levels=10, alpha=0.6)
+
+        plt.show()
+    """
+
+    __slots__ = ["frame", "lon_0", "lat_0", "sigma_semi_major", "e", "theta"]
+
+    def __init__(self, lon_0, lat_0, sigma_semi_major, e, theta, frame="galactic"):
+        self.frame = frame
+        self.lon_0 = Parameter(
+            "lon_0", Longitude(lon_0).wrap_at("180d"), min=-180, max=180
```

Comment:
Why wrap lon? Suggest to remove this (here and for SkyEllipse) unless you have a test that shows that it's needed.
The code should work both at lon of 0 deg and at 180 deg, no?

Reply:
See [this comment](https://github.com/gammapy/gammapy/pull/2066#issue-257735820)

Reply:
OK, fine, then leave the wrap_at as-is.

I think the proper solution is to never wrap, not at 0, not at 180, not at 360, and to only use sky separation instead of `lon1 - lon2`, and all will be fine.

But let's not do this discussion here.